app/dashboard/models.py

```diff
@@ -2236,6 +2236,24 @@ def to_dict(self, activities=True, leaderboards=True, network=None, tips=True):
 
         return params
 
+    @property
+    def locations(self):
+        from app.utils import get_location_from_ip
+        locations = []
+        try:
+            for login in self.actions.filter(action='Login'):
+                if login.location_data:
+                    locations.append(login.location_data)
+                else:
+                    location_data = get_location_from_ip(login.ip_address)
+                    login.location_data = location_data
+                    login.save()
+                    locations.append(location_data)
+        except Exception:
```

Comment:
Any idea what exception typically occurs here?

Reply:
Good spot.

The try/catch is from the original PR and it looks like, with the refactored code and with the [try/catch in `get_location_from_ip`](https://github.com/gitcoinco/web/blob/master/app/app/utils.py#L317-L324) it's not needed here. Will tidy that up.